lib/src/main/java/graphql/nadel/validation/NadelHydrationValidation.kt

```diff
@@ -120,30 +121,31 @@ internal class NadelHydrationValidation(
         hydrations: List<UnderlyingServiceHydration>,
     ): List<NadelSchemaValidationError> {
         if (hydrations.size > 1) {
-            val anyListSourceInputField = hydrations
-                .any { hydration ->
-                    val parentType = parent.underlying as GraphQLFieldsContainer
+            val pathsToSourceFields = hydrations
+                .asSequence()
+                .flatMap { hydration ->
                     hydration
                         .arguments
                         .asSequence()
-                        .mapNotNull { argument ->
-                            argument.remoteArgumentSource.pathToField
-                        }
-                        .any { path ->
-                            parentType.getFieldAt(path)?.type?.unwrapNonNull()?.isList == true
+                        .mapNotNull { it.remoteArgumentSource.pathToField }
+                }
+                .toList()
+
+            val parentType = parent.underlying as GraphQLFieldsContainer
+            val anyListSourceInputField = pathsToSourceFields
+                .any { pathToField ->
+                    parentType
+                        .getFieldsAlong(pathToField)
+                        .any { field ->
+                            field.type.unwrapNonNull().isList
```

Comment:
This is the actual change.

e.g. for `$source.contexts.userId` where `contexts` is a list but `userId` is not a list

Previously we only checked the end field to see if it was a list, this checks everything.